src/System.Text.Json/src/System/Text/Json/Serialization/Converters/DefaultConverters.cs

```diff
@@ -2,75 +2,68 @@
-            new JsonValueConverterDoubleNullable(),
-            new JsonValueConverterDecimalNullable(),
-            new JsonValueConverterDateTimeNullable(),
-            null,
-            new JsonValueConverterString()
-        };
-
-        internal static object GetDefaultPropertyValueConverter(Type propertyType, bool isNullable)
-        {
-            object converter = null;
-
-            int typeCode = (int)Type.GetTypeCode(propertyType);
-            if (typeCode <= MaxTypeCode)
+            if (type.IsEnum)
             {
-                if (isNullable)
-                {
-                    converter = s_NullableConverters[typeCode];
-                }
-                else
-                {
-                    converter = s_Converters[typeCode];
-                }
+                return Activator.CreateInstance(
+                    typeof(JsonValueConverterEnum<>).MakeGenericType(type),
+                    BindingFlags.Instance | BindingFlags.NonPublic,
+                    binder: null,
+                    new object[] { false },
+                    culture: null);
             }
-
-            return converter;
+            else if (type == typeof(Enum))
+            {
+                return new JsonValueConverterEnum(false);
+            }
+            TypeCode typeCode = Type.GetTypeCode(type);
+            switch (typeCode)
+            {
+                case TypeCode.Boolean:
+                    return new JsonValueConverterBoolean();
+                case TypeCode.Char:
+                    return new JsonValueConverterChar();
+                case TypeCode.SByte:
+                    return new JsonValueConverterSByte();
+                case TypeCode.Byte:
+                    return new JsonValueConverterByte();
+                case TypeCode.Int16:
+                    return new JsonValueConverterInt16();
+                case TypeCode.UInt16:
+                    return new JsonValueConverterUInt16();
+                case TypeCode.Int32:
+                    return new JsonValueConverterInt32();
+                case TypeCode.UInt32:
+                    return new JsonValueConverterUInt32();
+                case TypeCode.Int64:
+                    return new JsonValueConverterInt64();
+                case TypeCode.UInt64:
+                    return new JsonValueConverterUInt64();
+                case TypeCode.Single:
+                    return new JsonValueConverterSingle();
+                case TypeCode.Double:
+                    return new JsonValueConverterDouble();
+                case TypeCode.Decimal:
+                    return new JsonValueConverterDecimal();
+                case TypeCode.DateTime:
+                    return new JsonValueConverterDateTime();
+                case TypeCode.String:
+                    return new JsonValueConverterString();
+            }
+            return null;
         }
     }
+
+    internal static class DefaultConverters<TValue>
```

Comment:
The previous version of this class was returning a (globally) cached instance of a converter, meaning there'd be only one converter for a `decimal`, say.  
Am I correct in thinking that this change means that there's now a new converter for each `PropertyInfo` instance?  So larger JSON classes have much larger converter trees?  Or are things being cached somewhere else, and I just don't have enough insight into this namespace?  Or was there some other benefit gained from removing this type of caching?

Reply:
It's cached in the static `s_converter` field now. This was done to support caching generic enum converters.

Reply:
Argh, of course.  Because the class itself will be cached on first instantiation.